### DISCUSSION TITLE
How to use cookie jar

### DISCUSSION THREAD

**Discussion by ventidue** [nodejs/undici#864] [Q&A]
How can I make my sequence of requests use the cookies from the past request is there a cookie jar for undici or can I use something like tough cookie?

**Comment by mcollina** [comment]
We don't add anything for this but it would be pretty amazing to add this kind of API (or link to another module implementing it). At this point this feature does not exists.

**Reply by vikasg603** [reply]
Hey, @mcollina it's been 1 year since this feature request.
can we expect this in near future?

**Reply by mcollina** [reply]
Would you be willing to contribute this feature yourself or fund its development?

**Reply by vikasg603** [reply]
Hey, @mcollina I can try for the contribution.
I will implement it from my end will add a feature request.

**Reply by mcollina** [reply]
Awesome!

**Reply by KhafraDev** [reply]
I have an old branch that implements a cookie store at https://github.com/khafradev/undici/tree/cookie-store

**Reply by vikasg603** [reply]
hey, @KhafraDev are you going to continue development on it and push it on merge, or should I pick it from my end?
I will have to go through your code to understand it, which can take some time.

**Reply by KhafraDev** [reply]
I'd be willing to continue working on it if there would be a chance of it being implemented[.](https://github.com/nodejs/undici/pull/1441#issuecomment-1126977636) It was an implementation of the wicg [CookieStore](https://wicg.github.io/cookie-store/) which doesn't account for multiple origins, and likely isn't the best option in a server environment.

Not only that, it still required a lot of work to integrate into something useful, but can't be implemented directly into fetch without going against other implementations. It would also require re-working the Headers implementation to either handle `set-cookie` headers differently or not to instantly join duplicate headers on append, and do so on a lazy basis instead. Once the headers are concatenated, the CookieStore api can no longer handle them.

On top of that the code wasn't *great*, it still needs to enable WPTs (if they exist?), webidl, and probably a lot more I forgot. It's hard to work around these issues

**Reply by vikasg603** [reply]
Okay, @KhafraDev got the issues, I can understand how hard is to merge something while keeping the standard not affecting it.
I will still look into it and will write down the problems associated and let's see how it goes.
